### PR TITLE
GEODE-6027: refactor updateClusterConfig

### PR DESCRIPTION
**Pull request by pdxrunner**
- Add ResultModel parameter
- change from void return to boolean

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jdeppe-pivotal** [comment]
@aditya87 @petahhh FYI on this PR
Although it might seem pointless to separate the `ResultModel` and `configObject` as params, my feeling is that these don't belong together and the only reason `RM` contains the `configObject` is because the commands return a `RM`. To that end, I'm also considering having commands return a `Pair<ResultModel, Object>` (http://commons.apache.org/proper/commons-lang/javadocs/api-3.8/index.html?org/apache/commons/lang3/tuple/Pair.html) allowing us to separate the two concerns.  Thoughts?

File: geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java [hunk 1]
```diff
@@ -78,7 +78,8 @@ public ResultModel createMapping(
   }
 
   @Override
-  public void updateClusterConfig(String group, CacheConfig cacheConfig, Object element) {
+  public boolean updateClusterConfig(String group, CacheConfig cacheConfig, Object element,
+      ResultModel resultModel) {
```
Comment (jinmeiliao) [inline geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java]: is this resultModel used in any implementation?

File: geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java [hunk 2]
```diff
@@ -97,5 +98,6 @@ public void updateClusterConfig(String group, CacheConfig cacheConfig, Object el
         .stream()
         .filter(regionConfig -> regionConfig.getName().equals(newCacheElement.getRegionName()))
         .forEach(regionConfig -> regionConfig.getCustomRegionElements().add(newCacheElement));
+    return true;
```
Comment (jinmeiliao) [inline geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java]: where is the return value used?

File: geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java (hunk 1 above)
Comment (pdxrunner) [reply geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java]: Not yet. We wanted to due the refactoring of `updateClusterConfig` as an initial PR. Individual commands will use this as they are refactored to use `SingleGfshCommand`. Initial work on refactoring `AlterAsyncEventQueue` is what prompted this change.

File: geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java (hunk 2 above)
Comment (pdxrunner) [reply geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java]: The return value is used in  `CommandExecutor.invokeCommand`. Individual command classes that override `updateClusterConfig` will return `true` or `false` to indicate whether the configuration was updated or not (or possibly throw an exception if an error occurred updating the configuration).

**Comment by pivotal-jbarrett** [comment]
@pdxrunner is this PR going to see any more effort or should it be closed?

**Comment by pdxrunner** [comment]
> @pdxrunner is this PR going to see any more effort or should it be closed?

Thanks for pinging me on this. I should have closed it out when I superseded it with https://github.com/apache/geode/pull/2846

**Pull request closed without merging**